File: idea/testData/hierarchy/calls/callers/javaMethod/main0.java
class KBase {
    public String foo(String s) {
        return s;
    }
}

class KA extends KBase {
    public final String name = "A";

    @Override
    public final String <caret>foo(String s) {
        return "A " + s;
    }
}

class KClient {
    {
        new KBase().foo("");
        new KA().foo("");
    }

    public static final String a = new KBase().foo("") + new KA().foo("");

    public final String getBar() {
        return new KBase().foo("") + new KA().foo("");
    }

    public final String bar() {
        return new KBase().foo("") + new KA().foo("");
    }
}

File: kotlin-ide/idea/testData/hierarchy/calls/callers/kotlinProperty/main0.kt
open class KBase {
    open var name = ""
}

class KA: KBase() {
    override var <caret>name = "A"
    fun foo(s: String): String = "A: $s"
}

fun packageFun(s: String): String = KBase().name + KA().name

val packageVal = KBase().name + KA().name

class KClient {
    init {
        KBase().name = ""
        KA().name = ""
    }

    companion object {
        val a = KBase().name + KA().name
    }

    var bar: String
        get() = KBase().name + KA().name
        set(value: String) {
            KBase().name = value
            KA().name = value
        }

    fun bar() {
        fun localFun() = KBase().name + KA().name

        val s = KBase().name + KA().name
    }
}

object KClientObj {
    val a = KBase().name + KA().name
    init {
        KBase().name = ""
        KA().name = ""
    }
}